File: src/com/fusenetworks/fuse/util/PlayerData.java
package com.fusenetworks.fuse.util;

import java.util.HashMap;
import java.util.Map;
import java.util.UUID;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;

public class PlayerData {
    public static final Map<String, PlayerData> PLAYER_DATA = new HashMap<>();
    public static final long AUTO_PURGE = 20L * 60L * 5L;
    public static boolean hasPlayerData(Player player) {
        return PLAYER_DATA.containsKey(player.getPlayer().getAddress().getAddress().getHostAddress().trim());
    }
    public static PlayerData getPlayerDataSync(Player player) {
        synchronized (PLAYER_DATA) {
            return getPlayerData(player);
        }
    }
    public static PlayerData getPlayerData(Player player) {
        final String ip = player.getPlayer().getAddress().getAddress().getHostAddress().trim();
        PlayerData data = PlayerData.PLAYER_DATA.get(ip);
        if (data != null) {
            return data;
        }
        if (Bukkit.getOnlineMode()) {
            for (PlayerData dataTest : PLAYER_DATA.values()) {
                if (dataTest.player.getName().equalsIgnoreCase(player.getName())) {
                    data = dataTest;
                    break;
                }
            }
        }
        if (data != null) {
            return data;
        }
        data = new PlayerData(player, player.getUniqueId(), ip);
        PlayerData.PLAYER_DATA.put(ip, data);
        return data;
    }
    private final Player player;
    private final String ip;
    private final UUID uuid;
    private boolean cmdspyEnabled = false;
    public PlayerData(Player player, UUID uuid, String ip) {
        this.player = player;
        this.uuid = uuid;
        this.ip = ip;
    }
    public void setCommandSpy(boolean enabled) {
        this.cmdspyEnabled = enabled;
    }
    public boolean cmdspyEnabled() {
        return cmdspyEnabled;
    }
}

File: src/com/fusenetworks/fuse/util/PlayerData.kt
package com.fusenetworks.fuse.util


import java.util.HashMap
import java.util.UUID
import org.bukkit.Bukkit
import org.bukkit.entity.Player

class PlayerData(private val player: Player, private val uuid: UUID, private val ip: String) {
    private var cmdspyEnabled = false
    fun setCommandSpy(enabled: Boolean) {
        this.cmdspyEnabled = enabled
    }

    fun cmdspyEnabled(): Boolean {
        return cmdspyEnabled
    }

    companion object {
        val PLAYER_DATA: MutableMap<String, PlayerData> = HashMap()
        val AUTO_PURGE = 20L * 60L * 5L
        fun hasPlayerData(player: Player): Boolean {
            return PLAYER_DATA.containsKey(player.player.address.address.hostAddress.trim { it <= ' ' })
        }

        fun getPlayerDataSync(player: Player): PlayerData {
            synchronized(PLAYER_DATA) {
                return getPlayerData(player)
            }
        }

        fun getPlayerData(player: Player): PlayerData {
            val ip = player.player.address.address.hostAddress.trim { it <= ' ' }
            var data: PlayerData? = PlayerData.PLAYER_DATA[ip]
            if (data != null) {
                return data
            }
            if (Bukkit.getOnlineMode()) {
                for (dataTest in PLAYER_DATA.values) {
                    if (dataTest.player.name.equals(player.name, ignoreCase = true)) {
                        data = dataTest
                        break
                    }
                }
            }
            if (data != null) {
                return data
            }
            data = PlayerData(player, player.uniqueId, ip)
            PlayerData.PLAYER_DATA.put(ip, data)
            return data
        }
    }
}